### PR TITLE
translate to arabic

### PR DESCRIPTION
**Pull request by HusamJubran**

### CODE REVIEW

**Comment by pyphilia** [comment]
It looks good to me, I'll let you:
- add a language selector
- make the direction dynamic given the language
- clean comments

File: public/index.html [hunk 1]
```diff
@@ -47,6 +47,7 @@
 
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
+    <!--<script>document.dir = 'rtl';</script>-->
```
Comment (pyphilia) [inline public/index.html:50]: Remove this?

File: src/components/Root.tsx [hunk 2]
```diff
@@ -1,30 +1,75 @@
...
+            {/*/!*@ts-ignore*!/*/}
+            {SHOW_NOTIFICATIONS && <ToastContainer />}
+            <Select
+              label="Change Language"
+              value={curr_lang}
+              onChange={handleLangSelect}
+            >
+              <MenuItem value={'en'}>{'en'}</MenuItem>
+              <MenuItem value={'ar'}>{'ar'}</MenuItem>
+            </Select>
```
Comment (pyphilia) [inline src/components/Root.tsx:64]: As mentioned, use `@graasp/translations`'s `langs` to get all the available languages.

File: src/components/Root.tsx [hunk 3]
```diff
@@ -1,30 +1,75 @@
...
+const cacheLtr = createCache({
+  key: 'muiltr',
+  stylisPlugins: [prefixer],
+});
+
+const ltrTheme = { ...theme, direction: 'ltr' };
+const rtlTheme = { ...theme, direction: 'rtl' };
+
+const Root = () => {
+  const [curr_lang, setCurr_lang] = React.useState('en');
```
Comment (pyphilia) [inline src/components/Root.tsx:39]: In JS we use camel case so `currentLang` and `setCurrentLang`

File: src/components/Root.tsx [hunk 4]
```diff
@@ -1,30 +1,75 @@
...
+  stylisPlugins: [prefixer, rtlPlugin],
+});
+
+const cacheLtr = createCache({
+  key: 'muiltr',
+  stylisPlugins: [prefixer],
+});
+
+const ltrTheme = { ...theme, direction: 'ltr' };
+const rtlTheme = { ...theme, direction: 'rtl' };
```
Comment (pyphilia) [inline src/components/Root.tsx:36]: I think you can use `isRtl` to define one theme (define in the component)

File: src/components/Root.tsx [hunk 5]
```diff
@@ -1,30 +1,75 @@
...
-);
+const cacheRtl = createCache({
+  key: 'muirtl',
+  stylisPlugins: [prefixer, rtlPlugin],
+});
+
+const cacheLtr = createCache({
+  key: 'muiltr',
+  stylisPlugins: [prefixer],
+});
```
Comment (pyphilia) [inline src/components/Root.tsx:33]: Define one cache given `isRtl` (in the component)

File: src/components/Root.tsx [hunk 6]
```diff
@@ -1,30 +1,75 @@
...
+const Root = () => {
+  const [curr_lang, setCurr_lang] = React.useState('en');
+
+  const [isRtl, setIsRtl] = React.useState(true);
+
+  const handleLangSelect = (event) => {
+    setIsRtl(curr_lang === 'ar');
+    isRtl ? i18n.changeLanguage('ar') : i18n.changeLanguage('en');
+    isRtl ? (document.dir = 'rtl') : (document.dir = 'ltr');
+    setCurr_lang(event.target.value);
```
Comment (pyphilia) [inline src/components/Root.tsx:47]: Normally you can get the direction with `i18n.dir()`

**Review by pyphilia — CHANGES_REQUESTED** [review]
Thanks for the PR, see my comments below.

**Comment by pyphilia** [comment]
@spaenleh I've taken over this PR, you can review it <3

File: package.json [hunk 7]
```diff
@@ -28,7 +30,10 @@
     "react-i18next": "11.16.11",
     "react-router": "6.14.2",
     "react-router-dom": "6.14.2",
+    "react-scripts": "5.0.1",
```
Comment (spaenleh) [inline package.json:33]: remove

File: src/typings.d.ts
Comment (spaenleh) [inline src/typings.d.ts:1]: What is `stylis` used for ?

**Review by spaenleh — CHANGES_REQUESTED** [review]
Nice ! I have left some comments. Not sure why some packages were added. 

Can you provide a screenshot of the footer with the language select that would be nice 🖼️

File: package.json [hunk 8]
```diff
@@ -9,15 +9,17 @@
     "Alexandre Chau"
   ],
   "dependencies": {
+    "@emotion/cache": "11.10.7",
```
[the remark this answers is not in the document]
Comment (pyphilia) [reply package.json:12]: We need some packages for RTL support (hidden in graasp-ui)

**Comment by pyphilia** [comment]
Screenshot 2023-08-04 at 16 49 58
Screenshot 2023-08-04 at 16 50 09
The language select is a bit hidden because of react query aha

File: package.json (hunk 8 above)
[the remark this answers is not in the document]
Comment (spaenleh) [reply package.json:12]: Ah I see from [the docs](https://mui.com/material-ui/guides/right-to-left/) Nice !

**Review by spaenleh — APPROVED** [review]
Nice ! Just be careful as the language select might be just under the Captcha badge in the lower right corner 😅

**Pull request merged**